src/components/annotations/variants/VariantsView.vue

```diff
@@ -50,14 +52,18 @@ const unsubscribe = TextEventBus.on('click', ({ target }) => {
   } else {
     // if we are in single select mode, we still have variant annotations, but there are not shown
     // if we click at a part of text whose related annotations are not in the variant annotations, then we do not proceed further
-    const variantAnnotations = getVariantAnnotations(annotationStore.annotations, 'Variant')
-    if (!variantAnnotations.find((annotation) => annotation.id === ids[0])) {
-      return
-    }
+    let isAtLeastOneAnnotationTargetInVariants = false
+    ids.forEach((id) => {
+      if(variantAnnotationIds.includes(id))  isAtLeastOneAnnotationTargetInVariants = true
+    })
+    if(!isAtLeastOneAnnotationTargetInVariants) return
   }
 
   const targetIsSelected = parseInt(target.getAttribute('data-annotation-level'), 10) > 0
 
+  // ids are all the annotation ids of the target, which can be in different tabs. 
+  // we need to filter only the variant ides before proceeding with adding active annotation or removing active annotation
+  ids = ids.filter((id) => variantAnnotationIds.includes(id))
```

Comment:
Isn't this the same as the forEach above?

Reply:
Not really.  Each target is related with multiple types of annotations, like Editorial or Variant. I found a bug which calls addNewActiveAnnotation with an Editorial annotation. Therefore, we want to keep only the variant annotations for the code to follow.

Reply:
yes, but your filter in line 66 should be doing the same thing as lines 55 - 60, isn't it? you could then check also for ids.length > 0 instead of having isAtLeastOneAnnotationTargetInVariants.

Reply:
yes, true, I get your point. It becomes more compact.